### DISCUSSION TITLE
Optional QoL changes.

### DISCUSSION THREAD

**Discussion by LuisMenethil** [diasurgical/DevilutionX#2083] [Ideas]
Hello guys. First, thank you for the great work done here. I got tired of mods that change everything in the game and it turns into something else.  It's good to keep the experience closest to the original, adding and correcting just a few little things.

I would like to know if you plan to add a stash, and if you already have it, how can I enable it. About the item highlight on the floor, it's possible to make it similar to Diablo 2? You click Alt and see all the outlines and names. Or maybe an upgrade on Search spell, adding these features to it?

Lastly, in your Hellfire version the Mana Shield damage reduction bug and unique item drop chances bug are present?

**Comment by yuripourre** [comment]
@LuisMenethil item highlight was merged a few days ago (mostly based on @qndel's work), about the stash, @qndel also started something but is not complete.

I don't know how the search spell looks like nor the bugs you mentioned.

**Reply by LuisMenethil** [reply]
Thanks for the news. The bugs I mentioned are present in original Hellfire version 1.01. The Search spell works as a predecessor to the Alt function in Diablo 2. The difference is that only the outlines of the items are highlighted, the names aren't displayed. It's not very practical. As it is a spell, consumes mana for each use, again, not very practical.

**Reply by StephenCWills** [reply]
I've heard there are multiple competing stash implementations. I believe @qndel has dropped his and is currently favoring this one over his own: #1518.

**Reply by yuripourre** [reply]
@StephenCWills didn't know that. Thanks for the info!

**Comment by StephenCWills** [comment]
For Mana Shield, damage reduction is a flat 33% in DevilutionX. I'm not sure what the unique item drop chance bug is supposed to be. Maybe #628?

**Reply by LuisMenethil** [reply]
Thanks. 
This #540. It's the same?

**Reply by StephenCWills** [reply]
Well, it's a somewhat peculiar situation regarding the order of events and some of the details, but #628 is the issue and #540 is a pull request with a proposed fix. The discussion extends across both the issue and the PR, and probably Discord as well. The PR is actually referenced in the issue's first reply. As you can see, it's a rather complex problem that hasn't been fully solved yet.

**Comment by LuisMenethil** [comment]
Thank you guys for the info!

One last thing. I see a lot of people complaining about Hellfire that the Sorcerer starts the game with the skills already costing the minimum mana possible, unlike the original, where the mana cost starts at maximum and decreases as you read the skill books. Is this a bug or was it intended by the developers?

**Reply by AJenbo** [reply]
There where some bugs that caused the game to ignore spell level. They are fixed at this point afaik, you can try out the latest test build and see if what  you are referring to is solved.

**Reply by LuisMenethil** [reply]
Thanks again for the info. Where can I find the latest test build?

**Reply by qndel** [reply]
in readme 🙄

**Reply by LuisMenethil** [reply]
Okay. Sorry for my noobness.

**Comment by LuisMenethil** [comment]
Hi. I just tested the latest test build and the "bug" is still present. It only occurs on Hellfire.

This "bug" was reported on the GOG Forums:

https://www.gog.com/forum/general/diablo_hellfire_v101_mana_cost_bug

https://www.gog.com/forum/diablo/mana_cost_bug_in_hellfire

I managed to reproduce it:

Original Diablo without mana cost "bug", new Sorcerer hero:
https://ibb.co/KKb5VjM
Diablo Hellfire with mana cost "bug", new Sorcerer hero:
https://ibb.co/smwWX1K

Was it intended by the Hellfire developers?

**Reply by AJenbo** [reply]
They intentionally halfed the mana cost for all spells. But half is not the same as minimum.

**Reply by LuisMenethil** [reply]
Interesting. Strange idea since the Sorcerer was already super powerful.

**Reply by AJenbo** [reply]
There is an interview where they state that they felt he was very hard to play in the beginning and wanted to change that with this and other changes.

**Reply by LuisMenethil** [reply]
Curious. If the mana cost was intentionally halfed by the developers, why this only happens in Hellfire patch 1.01? I tested Hellfire without patch (1.00) and the mana cost remains the same as the original Diablo, no reductions. Are you sure it's not a bug introduced in Hellfire patch 1.01?

**Reply by AJenbo** [reply]
There is an interview where they state that they felt he was very hard to play in the beginning and wanted to change that with this and other changes.

**Comment by ropufu** [comment]
I've been thinking about the **Search** spell color-coding myself. In fact, the fix is pretty straightforward---I've implemented it in my local copy of **devilutionX**. And since I'm not the only one who thought about it, I will probably open a PR (once I'm done with the other PR I am currently working on).

Btw, the authors of **devilutionX** already have the `Alt` functionality built into the game. So the **Search** spell is---mildly put---unnecessary. I discovered it accidentally a couple of weeks ago :^)
